src/lib/aws/construct/static-asset-deployment/main.ts

```diff
@@ -14,49 +16,91 @@ import { StaticAssetDeploymentProps } from './types'
  * class CustomConstruct extends StaticAssetDeployment {
  *   constructor(parent: Construct, id: string, props: StaticAssetDeploymentProps) {
  *     super(parent, id, props)
- *     this.props = props
- *     this.id = id
  *     this.initResources()
  *   }
  * }
  */
-export class StaticAssetDeployment extends CommonConstruct {
-  /* construct properties */
-  props: StaticAssetDeploymentProps
-  id: string
-
+export class StaticAssetDeployment extends CommonConstruct<StaticAssetDeploymentProps> {
```

Comment:
Please raise a backlog JIRA for this change. We need to make changes consistently across the repository and should avoid different code defining props in the repository.